hexrd/ui/state_compatibility.py

```diff
@@ -0,0 +1,43 @@
+import h5py
+
+
+def update_if_needed(file_path):
+    # Find and fix any issues with the state file
+    fix_issue_1227(file_path)
+
+
+def has_issue_1227(file_path):
+    # Issue 1227 is where `\` was used on Windows for paths inside the HDF5
+    # file instead of `/`
+    with h5py.File(file_path, 'r') as rf:
+        return any(x.startswith('config\\') for x in rf.keys())
+
+
+def fix_issue_1227(file_path):
+    if not has_issue_1227(file_path):
+        return
+
+    print('State file found to contain issue #1227. Fixing it up...')
```

Comment:
We should probably try to switch all our `print`'s  to log statement sometime, I know we have discussed this.